crates/typst-library/src/visualize/image/svg.rs

```diff
@@ -287,3 +311,124 @@ impl FontResolver<'_> {
+            {
+                return Err(EcoString::from("URLs are not allowed"));
+            }
+        }
+
+        // Resolve the path to the linked image.
+        if self.svg_parent.is_none() {
+            return Err(EcoString::from("cannot access file system from here"));
+        }
+        // Replace the final underscore in svg_parent by the href.
+        let href_file = self.svg_parent.unwrap().join(href);
+
+        // Load image if file can be accessed.
+        match self.world.file(href_file) {
+            Ok(bytes) => {
+                let arc_data = Arc::new(bytes.to_vec());
+                let format = match determine_format_from_path(href) {
+                    Some(format) => Some(format),
+                    None => ImageFormat::detect(&arc_data),
+                };
+                match format {
+                    None => Err(EcoString::from("could not determine image format")),
+                    Some(ImageFormat::Vector(vector_format)) => match vector_format {
+                        VectorFormat::Svg => {
+                            Err(EcoString::from("SVG images are not supported yet"))
+                        }
+                        VectorFormat::Pdf => {
+                            Err(EcoString::from("PDF documents are not supported"))
+                        }
+                    },
+                    Some(ImageFormat::Raster(raster_format)) => match raster_format {
+                        RasterFormat::Exchange(exchange_format) => {
+                            match exchange_format {
+                                ExchangeFormat::Gif => Ok(usvg::ImageKind::GIF(arc_data)),
+                                ExchangeFormat::Jpg => {
+                                    Ok(usvg::ImageKind::JPEG(arc_data))
+                                }
+                                ExchangeFormat::Png => Ok(usvg::ImageKind::PNG(arc_data)),
+                                ExchangeFormat::Webp => {
+                                    Ok(usvg::ImageKind::WEBP(arc_data))
+                                }
+                            }
+                        }
+                        RasterFormat::Pixel(_) => {
+                            Err(EcoString::from("pixel formats are not supported"))
+                        }
+                    },
+                }
+            }
+            Err(err) => Err(match err {
+                FileError::NotFound(path) => {
+                    eco_format!("file not found, searched at {}", path.display())
+                }
+                FileError::AccessDenied => EcoString::from("access denied"),
+                FileError::IsDirectory => EcoString::from("is a directory"),
+                FileError::Other(Some(msg)) => msg,
+                FileError::Other(None) => EcoString::from("unspecified error"),
+                _ => EcoString::from("unexpected error"),
+            }),
```

Comment:
It's a bit unfortunate that this is duplicated here. There's probably some way to make that nicer, but I guess we can also defer that to later.

Reply:
In principle, the code can be simplified to something like

```rust
Err(err) => Err(err.to_string().into())
```

The downside is that the error messages become more convoluted. E.g.:

With one-liner, I get:

```text
failed to load linked image ../small.jpg in SVG (failed to load file (access denied))
```

With the match expression, this is made more compact:

```text
failed to load linked image ../small.jpg in SVG (access denied)
```

I considered transforming the output of `Err(err.to_string().into())` into a more condensed string, but that also requires a match expression to handle different cases, which seemed worse.

Reply:
Maybe add a
```
// TODO: Somehow unify this with `impl Display for FileError`.
```